sdk/core/Azure.Core/src/Shared/AzureSasCredentialPolicy.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal class AzureSasCredentialPolicy : HttpPipelineSynchronousPolicy
+    {
+        private readonly AzureSasCredential _credential;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="AzureSasCredentialPolicy"/> class.
+        /// </summary>
+        /// <param name="credential">The <see cref="AzureSasCredentialPolicy"/> used to authenticate requests.</param>
+        public AzureSasCredentialPolicy(AzureSasCredential credential)
+        {
+            Argument.AssertNotNull(credential, nameof(credential));
+            _credential = credential;
+        }
+
+        /// <inheritdoc/>
+        public override void OnSendingRequest(HttpMessage message)
+        {
+            base.OnSendingRequest(message);
```

Comment:
Nit - consider doing this last?  It feels like changing the URI should happen before whatever goes on in `base`, but it's a super nit thing because `base` is empty and probably always will be.